checker/src/main/java/org/checkerframework/checker/pico/jdk.astub

```diff
@@ -0,0 +1,355 @@
+import org.checkerframework.checker.pico.qual.Mutable;
+import org.checkerframework.checker.pico.qual.Immutable;
+import org.checkerframework.checker.pico.qual.ReceiverDependentMutable;
+import org.checkerframework.checker.pico.qual.Readonly;
+import org.checkerframework.checker.pico.qual.ObjectIdentityMethod;
+import java.util.Collection;
+
+package java.lang;
+
+@ReceiverDependentMutable
+class Iterable<E> {
+  Iterator<E> iterator(@Readonly Iterable this);
+}
+
+class AssertionError {
+  AssertionError(@Readonly Object var0);
+}
+
+class RuntimeException {
+  RuntimeException(String var0, Throwable var1);
+}
+
+@ReceiverDependentMutable
+class Object {
+  @ReceiverDependentMutable Object();
+  Class<?> getClass(@Readonly Object this);
+  String toString(@Readonly Object this);
+  int hashCode(@Readonly Object this);
+  boolean equals(@Readonly Object this, @Readonly Object var1);
+  @ReceiverDependentMutable Object clone(@ReceiverDependentMutable Object this);
+  @ObjectIdentityMethod
+  final native Class<?> getClass();
+}
+
+class String {
+  int length(@Immutable String this);
+  char charAt(@Immutable String this, int var1);
+  String replace(@Readonly CharSequence target, @Readonly CharSequence replacement);
+  boolean contains(@Readonly CharSequence s);
+  String substring(@Immutable String this, int var1);
+  String substring(@Immutable String this, int var1, int var2);
+  String toString(@Immutable String this);
+  boolean equals(@Immutable Object var1);
+  boolean contentEquals(@Readonly CharSequence var1);
+  static String valueOf(@Readonly Object var0);
+  static String format(String var0, @Readonly Object @Readonly ... var1);
+  static String format(@Readonly Locale l, String format, @Readonly Object @Readonly ... var1);
+}
+
+class StringBuilder {
+  StringBuilder append(@Readonly Object var1);
+}
+
+class StringBuffer {
```

Comment:
What default is currently used in stub files? If it's also immutable, many of these annotations are wrong.
In general, we should move to having two opt-in annotations, for a mutable or an immutable default, ASAP.

Reply:
Currently it is immutable. I have updated my annotations in the linked JDK PR.